src/core/server/saved_objects/types.ts

```diff
@@ -164,6 +165,7 @@ export interface SavedObjectsFindOptions {
 export interface SavedObjectsBaseOptions {
   /** Specify the namespace for this operation */
   namespace?: string;
+  requestOptions?: TransportRequestOptions;
```

Comment:
Oops, very good point indeed, we can't do that 🤔 

This is not for unit test purposes, but rather for proper shutdown of the Kibana server itself.
I imagine there's something we can do at SOR level, to cancel any ongoing HTTP requests if `stop()` is called, without having to expose the `signal` API to the callers.

I'll take a look and propose an alternative.

Reply:
I found a less impactful alternative:

Instead of cancelling the current `getOptInStatus()` request on `stop()`, what we do now is we wait for this request to finish. We also cancel the RxJS `timer()` and thus all the upcoming ones.